ivy/functional/ivy/nest.py

```diff
@@ -345,9 +402,33 @@ def set_nest_at_indices(
     >>> print(nest)
     ivy.array([[1., 11., 3.], [4., 5., 22.]])
     """
+    if shallow:
+        result = type(nest)(nest)
+    else:
+        result = copy_nest(nest)
+    outer_tupled = False
+    if isinstance(nest, tuple):
+        outer_tupled = True
+        result = list(result)
     if not isinstance(values, (list, tuple)):
         values = [values] * len(indices)
-    [set_nest_at_index(nest, index, value) for index, value in zip(indices, values)]
+    for i, (index, value) in enumerate(zip(indices, values)):
+        if i == 0:
+            result = set_nest_at_index(
+                nest, index, value, _result=result, shallow=shallow
+            )
+            continue
+        if shallow:
+            result = set_nest_at_index(
+                nest, index, value, _result=result, shallow=shallow
+            )
+        else:
+            result = set_nest_at_index(
+                result, index, value, _result=result, shallow=shallow
+            )
```

Comment:
maybe I'm missing something, but can't we replace this block with just `result = set_nest_at_index(nest, index, value, _result=result, shallow=shallow)`? with the try except block in the `set_nest_at_index` it should behave the same way I think

Reply:
If we do that then `set_nest_at_index` won't receive updated nest in every loop in case of `shallow=False`.
`result = set_nest_at_index(nest, index, value, _result=result, shallow=shallow)` will only end up updating on the last index provided. Thats why I'm passing `result` as input in case of  `shallow=False`

Reply:
resolving this as discussed in discord